### PR TITLE
Move enforce_ssl to production:enforce_tls

### PR DESCRIPTION
**Pull request by mike-burns**
Move the `suspenders:enforce_ssl` generator to
`suspenders:production:enforce_tls`. This namespaces it as something
specific to production (useful for those who are looking for the
Heroku-specific functionality), and it renames SSL to TLS, as that is
the abbreviation used in this century.

While here, shuffle some things around, which exposes existing bugs.
And fix those bugs!

### CODE REVIEW

File: lib/suspenders/generators/production/email_generator.rb [hunk 1]
```diff
@@ -4,15 +4,15 @@ module Suspenders
...
+        File.join("..", "..", "..", "..", "templates"),
         File.dirname(__FILE__),
       )
 
       def smtp_configuration
         copy_file "smtp.rb", "config/smtp.rb"
 
         prepend_file "config/environments/production.rb",
-          %{require Rails.root.join("config/smtp")\n\n}
+          %{require Rails.root.join("config/smtp")\n}
```
Comment (mike-burns) [inline lib/suspenders/generators/production/email_generator.rb]: We, unfortunately, have a test that ensures our `production.rb` has no vertical whitespace in it.

**Comment by jferris** [comment]
This makes me twitchy because Rails calls it `enforce_ssl`. That means the generator name doesn't match the setting name. On the other hand, the setting name doesn't match the protocol! What's a dev to do?

**Comment by mike-burns** [comment]
I've been trying to name the generators after what they do and not how it is implemented. For example, the `suspenders:lint` generator configures Hound, but if we want to change from Hound to Code Climate in the future we won't need to rename it. Similarly `suspenders:js_driver` (c-w), `suspenders:testing` (Rspec), `suspenders:analytics` (Segment), etc.

Therefore, maybe `suspenders:force_tls` is shorter and nicer. Eh?

**Comment by jferris** [comment]
That makes sense and I feel better now.

**Pull request merged**